app/src/main/java/ch/epfl/sdp/contamination/CachingDataSender.java

```diff
@@ -40,4 +47,31 @@ void registerLocation(Carrier carrier,
                           Date time,
                           OnSuccessListener successListener,
                           OnFailureListener failureListener);
+    /**
+     * Notifies a user he has been close to an infected person
+     */
+    default void sendAlert(String userId){
+        sendAlert(userId,0f);
+    }
+
+    /**
+     * If yesterday you were 90% surely sick, the person you met were already aware that they might be infected.
+     * Thus, it is "less" scary for them to know you are know sick. This method implements just that.

```

Comment:
Do you mean "if someday you are 90% surely sick, we notify the persons you met that they met a 90% surely infected person" ?

Reply:
No. This is done on the normal contamination detection part. My branch is only about what happens when a user clicks on the button to notify he became sick.

Reply:
Ok, from what I understand by reading your code again, I guess this function is not handling any particular case where some user was 90% surely sick, but it notifies users you met so that they know they might be the one that infected you.

Reply:
Yes approximatively. Let's have a skype to explain this.